src/main/java/com/gitblit/wicket/pages/MyTicketsPage.java

```diff
@@ -52,355 +57,442 @@
-			final int page,
-			int pageSize,
-			int count,
-			int total) {
-
-		boolean showNav = total > (2 * pageSize);
-		boolean allowPrev = page > 1;
-		boolean allowNext = (pageSize * (page - 1) + count) < total;
-		add(new BookmarkablePageLink<Void>("prevLink", MyTicketsPage.class, queryParameters(query, milestone, states, assignedTo, sort, desc, page - 1)).setEnabled(allowPrev).setVisible(showNav));
-		add(new BookmarkablePageLink<Void>("nextLink", MyTicketsPage.class, queryParameters(query, milestone, states, assignedTo, sort, desc, page + 1)).setEnabled(allowNext).setVisible(showNav));
-
-		if (total <= pageSize) {
-			add(new Label("pageLink").setVisible(false));
-			return;
-		}
-
-		// determine page numbers to display
-		int pages = count == 0 ? 0 : ((total / pageSize) + (total % pageSize == 0 ? 0 : 1));
-		// preferred number of pagelinks
-		int segments = 5;
-		if (pages < segments) {
-			// not enough data for preferred number of page links
-			segments = pages;
-		}
-		int minpage = Math.min(Math.max(1, page - 2), pages - (segments - 1));
-		int maxpage = Math.min(pages, minpage + (segments - 1));
-		List<Integer> sequence = new ArrayList<Integer>();
-		for (int i = minpage; i <= maxpage; i++) {
-			sequence.add(i);
-		}
-
-		ListDataProvider<Integer> pagesDp = new ListDataProvider<Integer>(sequence);
-		DataView<Integer> pagesView = new DataView<Integer>("pageLink", pagesDp) {
-			private static final long serialVersionUID = 1L;
-
-			@Override
-			public void populateItem(final Item<Integer> item) {
-				final Integer i = item.getModelObject();
-				LinkPanel link = new LinkPanel("page", null, "" + i, MyTicketsPage.class, queryParameters(query, milestone, states, assignedTo, sort, desc, i));
-				link.setRenderBodyOnly(true);
-				if (i == page) {
-					WicketUtils.setCssClass(item, "active");
-				}
-				item.add(link);
-			}
-		};
-		add(pagesView);
-	}
+    public MyTicketsPage() {
+        this(null);
+    }
+
+    public MyTicketsPage(PageParameters params) {
+        super(params);
+        setupPage("", getString("gb.myTickets"));
+
+        UserModel currentUser = GitBlitWebSession.get().getUser();
+        if (currentUser == null || UserModel.ANONYMOUS.equals(currentUser)) {
+            setRedirect(true);
+            setResponsePage(getApplication().getHomePage());
+            return;
+        }
+
+        final String username = currentUser.getName();
+        final String defaultQueryParam = "watchedby:"+username;
+
+        final String[] statiiParam = (params == null) ? TicketsUI.openStatii : params.getStringArray(Lucene.status.name());
+        final String assignedToParam = (params == null) ? "" : params.getString(Lucene.responsible.name(), null);
+        final String milestoneParam = (params == null) ? "" : params.getString(Lucene.milestone.name(), null);
+        final String queryParam = (params == null || StringUtils.isEmpty(params.getString("q", null))) ? defaultQueryParam : params.getString("q", null);
```

Comment:
BTW, there seems to be a bug in the mention detection. The mention is added to the list only if @username appears in the middle of the comment. If the comment starts with @username, it is not detected. Looks like it can be fixed with tweaking the regex.

Also, shouldn't the main ticket body also be processed for mentions instead of just comments?